packages/@aws-cdk/aws-pipes-sources-alpha/lib/kinesis.ts

```diff
@@ -0,0 +1,161 @@
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-parallelizationfactor
+   * @default 1
+   */
+  readonly parallelizationFactor?: number;
+
+  /**
+   * (Streams only) The position in a stream from which to start reading.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingposition
+   */
+  readonly startingPosition: KinesisStartingPosition;
+
+  /**
+   * With StartingPosition set to AT_TIMESTAMP, the time from which to start reading, in Unix time seconds.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingpositiontimestamp
+   * @default no starting position timestamp
+   */
+  readonly startingPositionTimestamp?: string;
+}
+
+/**
+ * A source that reads from Kinesis.
+ */
+export class KinesisSource implements ISource {
+  private readonly stream: IStream;
+  readonly sourceArn;
+  private sourceParameters;
+
+  private batchSize;
+  private maximumBatchingWindowInSeconds;
+  private maximumRecordAgeInSeconds;
+  private maximumRetryAttempts;
+  private parallelizationFactor;
+
+  constructor(stream: IStream, parameters: KinesisSourceParameters) {
+    this.stream = stream;
+    this.sourceArn = stream.streamArn;
+    this.sourceParameters = parameters;
+
+    this.batchSize = this.sourceParameters.batchSize;
+    this.maximumBatchingWindowInSeconds = this.sourceParameters.maximumBatchingWindow?.toSeconds();
+    this.maximumRecordAgeInSeconds = this.sourceParameters.maximumRecordAge?.toSeconds();
+    this.maximumRetryAttempts = this.sourceParameters.maximumRetryAttempts;
+    this.parallelizationFactor = this.sourceParameters.parallelizationFactor;
+
+    if (this.batchSize !== undefined) {
+      if (this.batchSize < 1 || this.batchSize > 10000) {
+        throw new Error(`Batch size must be between 1 and 10000, received ${this.batchSize}`);
+      }
+    }
+    if (this.maximumBatchingWindowInSeconds !== undefined) {
+      // only need to check upper bound since Duration amounts cannot be negative
+      if (this.maximumBatchingWindowInSeconds > 300) {
+        throw new Error(`Maximum batching window must be between 0 and 300, received ${this.maximumBatchingWindowInSeconds}`);
+      }
+    }
+    if (this.maximumRecordAgeInSeconds !== undefined) {
+      // only need to check upper bound since Duration amounts cannot be negative
+      if (this.maximumRecordAgeInSeconds > 604800) {
+        throw new Error(`Maximum record age in seconds must be between -1 and 604800, received ${this.maximumRecordAgeInSeconds}`);
+      }
+    }
```

Comment:
Same here, is 0 seconds valid?

Reply:
Good shout. Doesn't say it in the docs but needs to be min 60 if defined.